examples/apps/cli/Makefile.am

```diff
@@ -186,6 +78,7 @@ ot_cli_ftd_LDADD                                                       = \
     $(top_builddir)/src/core/libopenthread-ftd.a                         \
     $(LDADD_COMMON)                                                      \
     $(top_builddir)/src/core/libopenthread-ftd.a                         \
+    $(LDADD_COMMON)                                                      \
```

Comment:
nit: This and line above look like a duplication. Same for mtd and ncp.

Reply:
@LuDuda Good eye.  The list of libraries is duplicated to allow compilation to succeed even when circular dependencies exist.

Reply:
Is the second `$(LDADD_COMMON)` necessary?  Seems like the first three lines are sufficient.

Reply:
It is used to handle potential cyclical dependencies between platform and core.
I can remove if needed, but it is required for some future development in progress.